rows.go

```diff
@@ -27,6 +27,11 @@ var (
 	maxChunkDownloaderErrorCounter = 5
 )
 
+// SnowflakeRows provides an API for methods exposed to the clients
+type SnowflakeRows interface {
```

Comment:
While this makes sense, I think we can consolidate this into SnowflakeResult since this object can and should be able to return both a query ID and its status as well. Asking the user to convert this into separate interfaces seems cumbersome for both sides in terms of usability & maintenance.

Reply:
Sounds good. Felt a bit strange to have SnowflakeResult contain an endpoint related to data since Result is usually referred to by an `exec` query that does not return any data, but I agree it's better to expose only one interface. Related to that, would you prefer that the implementation of `GetBatches() ([]*ResultBatch, error)` by `snowflakeResult` returns an error, or returns an empty array (and nil)?

Reply:
I pushed it out with the former (returns an error), but happy to change it to empty array if you rather have that!

Reply:
Completely understandable - this SnowflakeResult object was meant to be detached from the snowflakeResult (related to Exec) however, and actually denote a literal result. Perhaps a confusing misnomer on our end.

I think the return value seems fine as is for now. I'll let you know if it needs changing or change it myself.